### PR TITLE
Update tests to use testify/require

### LINKED ISSUE #349: Update tests to use testify/require instead of value comparisons and Fatal
**Describe the problem/challenge you have**
Currently unit tests and e2e tests are using value comparisons and Fatal which is not very intuitive.

**Describe the solution you'd like**
We should update the tests to use testify/require which will make them more readable and the output would be very descriptive and concise in case the tests are failing.

**Anything else you would like to add:**
This has already been done for diff package, and the [PR](https://github.com/vmware-tanzu/carvel-kapp/pull/347) can be found here.
Update the tests for the following packages
- [x] pkg/kapp/diffgraph and pkg/kapp/matcher - @khan-ajamal
- [x] pkg/kapp/resources - @khan-ajamal
- [x] pkg/kapp/resourcemisc - @ShajithaMohammed
- [x] test/e2e - @ShajithaMohammed 

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### PR DESCRIPTION
**Pull request by ShajithaMohammed**
Update tests in **pkg/kapp/resourcemisc** to use testify/require instead of value comparisons and Fatal.
Fixes https://github.com/vmware-tanzu/carvel-kapp/issues/349

### CODE REVIEW

**Review by praveenrewar — APPROVED** [review]
LGTM

**Review by 100mik — APPROVED** [review]
Looks good! Thanks a lot.

File: pkg/kapp/resourcesmisc/apps_v1_deployment_test.go [hunk 1]
```diff
@@ -4,6 +4,7 @@
 package resourcesmisc_test
 
 import (
+	"github.com/stretchr/testify/require"
```
Comment (cppforlife) [inline pkg/kapp/resourcesmisc/apps_v1_deployment_test.go]: nit: lets move this import next to other github.com imports

Comment (100mik) [reply pkg/kapp/resourcesmisc/apps_v1_deployment_test.go]: @ShajithaMohammed could you make this change please? Then we can go ahead with a merge!

Comment (ShajithaMohammed) [reply pkg/kapp/resourcesmisc/apps_v1_deployment_test.go]: Sure, on it!

Comment (100mik) [reply pkg/kapp/resourcesmisc/apps_v1_deployment_test.go]: Thanks!

**Pull request merged**